hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStoreFile.java

```diff
@@ -227,8 +228,8 @@ public long getMaxMemStoreTS() {
    * @param primaryReplica true if this is a store file for primary replica, otherwise false.
    */
   public HStoreFile(FileSystem fs, Path p, Configuration conf, CacheConfig cacheConf,
-    BloomType cfBloomType, boolean primaryReplica) throws IOException {
-    this(new StoreFileInfo(conf, fs, p, primaryReplica), cfBloomType, cacheConf);
+    BloomType cfBloomType, boolean primaryReplica, StoreFileTracker sft) throws IOException {
```

Comment:
Rather than adding this extra 'sft' parameter, and change every single caller in the upper chain to now create/forward an SFT instance down here, can we simply create the SFT instance in this HStoreFile constructor?

Reply:
Currently HStoreFile is tied to Hadoop FileSystem obj, it will be more inline to send in the StoreFileTracker instance instead know?
Also then we need to send in the entire content of StoreContext to HStoreFile constructor if we want to create StoreFileTracker instance inside HStoreFile constructor.
Infact SFT instance currently doesnt have api for accessing its storeContext outside, otherwise we can replace FS object in constructor with SFT object directly

Reply:
How about building a StoreContext and passing that to the HStoreFile constructor? 
This could be a follow up refactor that cleans up this and more.